dev/tools/localization/gen_l10n.dart

```diff
@@ -163,6 +163,11 @@ String generateMethod(Message message, AppResourceBundle bundle) {
         messageValue = messageValue.replaceAll('{${placeholder.name}}', '\${${placeholder.name}}');
       }
     }
+
+    // Escape single and double quotes.
+    messageValue = messageValue.replaceAll("'", '\\\'');
```

Comment:
Shouldn't we be using generateString - https://github.com/flutter/flutter/blob/master/dev/tools/localization/localizations_utils.dart#L387

Reply:
That seems to be breaking integration tests. Investigating

Reply:
It seems to be generating raw strings for scenarios like:
```
  @override
  String helloOn(Object world, DateTime date, DateTime time) {
    final intl.DateFormat dateDateFormat = intl.DateFormat.yMd(_localeName);
    final String dateString = dateDateFormat.format(date);
    final intl.DateFormat timeDateFormat = intl.DateFormat.Hm(_localeName);
    final String timeString = timeDateFormat.format(time);

    return r"'Hello ${world} on ${dateString} at ${timeString}'";
  }
```

I could repurpose the generateString function to not do so

Reply:
Lets leave the PR as it is for now.